core-bundle/src/Resources/contao/library/Contao/Model.php

```diff
@@ -750,6 +750,13 @@ public function preventSaving($blnKeepClone=true)
 	 */
 	public static function findByPk($varValue, array $arrOptions=array())
 	{
+		if ($varValue === null)
+		{
+			trigger_deprecation('contao/core-bundle', '4.13', 'Using "%s::findByPk(null)" has been deprecated and will no longer work in Contao 5.0.', __CLASS__);
```

Comment:
Not sure if this is good. What if I want to pass `null` as argument to e.g. get `SELECT * FROM table WHERE field=null`?

Reply:
That’s why I only added it for `findByPk`, AFAIK we don’t support nullable primary key columns.

Reply:
IMHO nullable primary keys never make sense as it's semantically the same as deleting the row. :smile:

Reply:
But we also have `findById()` and `findByIdOrAlias()`. 🙈

Reply:
Yes, but `findByIdOrAlias()` could potentially allow `null` so we cannot disallow that I think.

Calling `findByPk(null)` is most certainly wrong, but for all other columns we cannot assume that.

Reply:
Not even for `findById()`? `findByPk()` is actually just an alias for `findById()`.

Reply:
Added deprecation for `findById()` in 325ad84cfa783bbe8f95c006f785da1db02f41b3